hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/table/catalog/TestHoodieHiveCatalog.java

```diff
@@ -0,0 +1,163 @@
+import org.apache.flink.table.catalog.exceptions.TableAlreadyExistException;
+import org.apache.flink.table.catalog.exceptions.TableNotExistException;
+import org.apache.flink.table.factories.FactoryUtil;
+import org.apache.hadoop.hive.metastore.api.Table;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.apache.flink.table.factories.FactoryUtil.CONNECTOR;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Test cases for {@link HoodieHiveCatalog}.
+ */
+public class TestHoodieHiveCatalog {
+  TableSchema schema =
+      TableSchema.builder()
+          .field("uuid", DataTypes.INT().notNull())
+          .field("name", DataTypes.STRING())
+          .field("age", DataTypes.INT())
+          .field("ts", DataTypes.BIGINT())
+          .field("par1", DataTypes.STRING())
+          .primaryKey("uuid")
+          .build();
+  List<String> partitions = Collections.singletonList("par1");
+  private static HoodieHiveCatalog hoodieCatalog;
+  private final ObjectPath tablePath = new ObjectPath("default", "test");
+
+  @BeforeAll
+  public static void createCatalog() {
+    hoodieCatalog = HoodieCatalogTestUtils.createHiveCatalog();
+    hoodieCatalog.open();
+  }
+
+  @AfterEach
+  public void dropTable() throws TableNotExistException {
+    hoodieCatalog.dropTable(tablePath, true);
+  }
+
+  @AfterAll
+  public static void closeCatalog() {
+    if (hoodieCatalog != null) {
+      hoodieCatalog.close();
+    }
+  }
+
+  @ParameterizedTest
+  @EnumSource(value = HoodieTableType.class)
+  public void testCreateAndGetHoodieTable(HoodieTableType tableType) throws Exception {
+    Map<String, String> originOptions = new HashMap<>();
+    originOptions.put(FactoryUtil.CONNECTOR.key(), "hudi");
+    originOptions.put(FlinkOptions.TABLE_TYPE.key(), tableType.toString());
+
+    CatalogTable table =
+        new CatalogTableImpl(schema, partitions, originOptions, "hudi table");
+    hoodieCatalog.createTable(tablePath, table, false);
+
+    CatalogBaseTable table1 = hoodieCatalog.getTable(tablePath);
+    assertEquals(table1.getOptions().get(CONNECTOR.key()), "hudi");
+    assertEquals(table1.getOptions().get(FlinkOptions.TABLE_TYPE.key()), tableType.toString());
+    assertEquals(table1.getOptions().get(FlinkOptions.RECORD_KEY_FIELD.key()), "uuid");
+    assertEquals(table1.getOptions().get(FlinkOptions.PRECOMBINE_FIELD.key()), "ts");
+    assertEquals(table1.getUnresolvedSchema().getPrimaryKey().get().getColumnNames(), Collections.singletonList("uuid"));
+    assertEquals(((CatalogTable)table1).getPartitionKeys(), Collections.singletonList("par1"));
+  }
```

Comment:
Should test for both external and managed table.

Reply:
testCreateExternalTable